demonstrations/tutorial_learning_few_data.py

```diff
@@ -0,0 +1,473 @@
+    Args:
+        weights (np.array): Array with the weights of the conditional U3 gate.
+        wires (list[int]): List of wires to apply the pooling layer on.
+    """
+    n_wires = len(wires)
+    assert len(wires) >= 2, "this circuit is too small!"
+
+    for indx, w in enumerate(wires):
+        if indx % 2 == 1 and indx < n_wires:
+            m_outcome = qml.measure(w)
+            qml.cond(m_outcome, qml.U3)(*weights, wires=wires[indx - 1])
+
+
+##############################################################################
+# We can construct a quantum CNN by combining both layers and using an arbitrary unitary to model
+# a dense layer. It will take as input a set of features (the image), encode these features using
+# an embedding map, apply rounds of convolutional and pooling layers, and eventually get the
+# desired measurement statistics of the circuit.
+
+def conv_and_pooling(kernel_weights, n_wires):
+    """Apply both the convolutional and pooling layer."""
+    convolutional_layer(kernel_weights[:15], n_wires)
+    pooling_layer(kernel_weights[15:], n_wires)
+
+
+def dense_layer(weights, wires):
+    """Apply an arbitrary unitary gate to a specified set of wires."""
+    qml.ArbitraryUnitary(weights, wires)
+
+num_wires = 6
+device = qml.device('default.qubit', wires=num_wires)
+
+@qml.qnode(device, interface="jax")
+def conv_net(weights, last_layer_weights, features):
+    """Define the QCNN circuit
+    Args:
+        weights (np.array): Parameters of the convolution and pool layers.
+        last_layer_weights (np.array): Parameters of the last dense layer.
+        features (np.array): Input data to be embedded using AmplitudEmbedding."""
+
+    layers = weights.shape[1]
+    wires = list(range(num_wires))
+
+    # inputs the state input_state
+    qml.AmplitudeEmbedding(features=features, wires=wires, pad_with=0.5)
+
+    # adds convolutional and pooling layers
+    for j in range(layers):
+        conv_and_pooling(weights[:, j], wires)
+        wires = wires[::2]
+
+    assert (
+            last_layer_weights.size == 4 ** (len(wires)) - 1
+    ), f"The size of the last layer weights vector is incorrect! \n Expected {4 ** (len(wires)) - 1}, Given {last_layer_weights.size}"
+    dense_layer(last_layer_weights, wires)
+    return qml.probs(wires=(0))
+
+
+##############################################################################
+# In the problem we will address, we need to encode 64 features
+# in the state to be processed by the QCNN. Thus, we require six qubits to encode
+# each feature value in the amplitude of each computational basis state.
+#
+# Training the QCNN on the digits dataset
+# ---------------------------------------
+# In this demo, we are going to classify the digits ``0`` and ``1`` from the classical ``digits`` dataset.
+# These are 8 by 8 pixel arrays of hand-written digits as shown below.
+
+digits = datasets.load_digits();
+digits = datasets.load_digits()
```

Comment:
why twice? Is there a reason why there is a semicolon? For suppressing the output?
```suggestion
digits = datasets.load_digits()
```